WordPress/Classes/ViewRelated/Notifications/Tools/NotificationMediaDownloader.swift

```diff
@@ -68,73 +133,112 @@ import Foundation
         let operation                   = AFHTTPRequestOperation(request: request)
         operation.responseSerializer    = responseSerializer
         operation.setCompletionBlockWithSuccess({
-            [weak self]
             (AFHTTPRequestOperation operation, AnyObject responseObject) -> Void in
             
             if let unwrappedImage = responseObject as? UIImage {
-                callback(nil, unwrappedImage)
+                success(unwrappedImage)
             }
             
+            self.beingDownloaded.remove(url)
         }, failure: {
             (AFHTTPRequestOperation operation, NSError error) -> Void in
-            callback(error, nil)
+            
+            self.beingDownloaded.remove(url)
         })
         
         downloadQueue.addOperation(operation)
-        retryMap[url] = retryCountForURL(url) + 1
+        beingDownloaded.insert(url)
+        increaseRetryCount(url)
     }
     
-    private func retryCountForURL(url: NSURL) -> Int {
+    /**
+    *  @brief       Checks if an image should be downloaded, or not.
+    *  @details     An image should be downloaded if:
+    *
+    *               -   It's not already being downloaded
+    *               -   Isn't already in the cache!
+    *               -   Hasn't exceeded the retry count
+    *
+    *  @param       urls            The collection of URL's of the assets you'd need.
+    *  @returns     A dictionary with URL as Key, and Image as Value.
+    */
+    private func shouldDownloadImage(#url: NSURL) -> Bool {
+        return originalImagesMap[url] == nil && checkRetryCount(url) < maximumRetryCount && !beingDownloaded.contains(url)
+    }
+    
+    /**
+    *  @brief       Increases the retry count for a given URL
+    *
+    *  @param       urls            The URL we're tracking
+    */
+    private func increaseRetryCount(url: NSURL) {
+        retryMap[url] = checkRetryCount(url) + 1
+    }
+
+    /**
+    *  @brief       Returns the current retry count for a given URL
+    *
+    *  @param       urls            The URL we're tracking
+    *  @return      The current retry count
+    */
+    private func checkRetryCount(url: NSURL) -> Int {
```

Comment:
Another very nitpicky comment, but... would `getRetryCount` be more descriptive since the method returns the count and any checking would be done by the caller?